examples/example_ds3_pp.py

```diff
@@ -253,57 +280,61 @@ def shape_inference_output_fn_last_stage():
             requires_grad=True,
         )
 
-    # Step 1. Construct the logical pipeline stages
-    with torch.device("meta"):
-        stage0 = DeepSeekV3Stage0(embed, layers[0], config)
-        stage1 = DeepSeekV3StageI(layers[1], config)
-        stage2 = DeepSeekV3StageI(layers[2], config)
-        stage3 = DeepSeekV3StageI(layers[3], config)
-        stage4 = DeepSeekV3StageI(layers[4], config)
-        stage5 = DeepSeekV3StageI(layers[5], config)
-        stage6 = DeepSeekV3StageI(layers[6], config)
-        stage7 = DeepSeekV3StageN(layers[7], norm, output, config)
-        logical_stages = [
-            stage0,
-            stage1,
-            stage2,
-            stage3,
-            stage4,
-            stage5,
-            stage6,
-            stage7,
-        ]
-    # Step 2. Assign each logical stage(s) to pp ranks
-    # This mapping is dependent of the number of logical pipeline stages, the pp_degree and the schedule
-    # For interleaved 1F1B, the mapping is:
-    # pp_rank_to_stage_indices = {
-    #     0: [0, 4],
-    #     1: [1, 5],
-    #     2: [2, 6],
-    #     3: [3, 7],
-    # }
-    # For DualPipeV, the mapping is:
-    # pp_rank_to_stage_indices = {
-    #     0: [0, 7],
-    #     1: [1, 6],
-    #     2: [2, 5],
-    #     3: [3, 4],
-    # }
-    pp_rank_to_stage_indices: dict[int, list[int]] = {
-        0: [0, 4],
-        1: [1, 5],
-        2: [2, 6],
-        3: [3, 7],
-    }
+    if fake_evaluate:
```

Comment:
is it important to maintain 2 paths one for 8 stage one for 4 stage? if not it'd be nice to clean it up and have only one codepath

Reply:
Unified the code path